data.md

```diff
@@ -8,17 +8,31 @@ module WASM-DATA
     imports DOMAINS
 ```
 
-Layout
-------
+Parsing
+-------
+
+### Layout
 
 WASM allows for block comments using `(;` and `;)`, and line comments using `;;`.
 Additionally, white-space is skipped/ignored.
+Declaring regular expressions of sort `#Layout` infroms the K lexer to drop these tokens.
+
+```k
+    syntax #Layout ::= r"\\(;([^;]|(;+([^;\\)])))*;\\)"
+                     | r";;[^\\n\\r]*"
+                     | r"[\\ \\n\\r\\t]"
+ // ------------------------------------
+```
+
+### Identifiers
+
+As defined in the [WASM Spec], the syntax of identifiers is as follows.
+
+**TODO**: Unsupported characters: `.:^@`
 
 ```k
-    syntax #Layout ::= r"(\\(;([^;]|(;+([^;\\)])))*;\\))"
-                     | r"(;;[^\\n\\r]*)"
-                     | r"([\\ \\n\\r\\t])"
- // --------------------------------------
+    syntax Identifier ::= r"\\$[0-9a-zA-Z!$%&'*+/<>?_`|~=-]*" [avoid, token]
+ // ------------------------------------------------------------------------
```

Comment:
One `-` too many.

Reply:
Hmmm, these show up as correct on mine. Can you double-check locally in your editor?

Reply:
Yep, they are correct.
Looks like GitHub contracts the `fi` in `Identifier` when displaying.